device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/amqps/AmqpsIotHubConnection.java

```diff
@@ -899,13 +913,16 @@ private void sendQueuedMessages()
                 AmqpsSessionHandler reconnectingDeviceSessionHandler = this.reconnectingDeviceSessionHandlers.get(message.getConnectionDeviceId());
                 if (reconnectingDeviceSessionHandler != null)
                 {
-                    log.trace("Amqp message failed to send because its AMQP session is currently reconnecting. Adding it back to messages to send queue ({})", message);
-                    messagesToSend.add(message);
```

Comment:
re-queueing the message that failed to send here skips consulting the retry policy. So instead, we need to kick the message back up a layer to IotHubTransport via the ```listener.onMessageSent()``` callback so that the retry policy gets to validate that a message should be resent.